File: backend/src/main/java/com/yangshengzhou/lucky_sms/pojo/Teacher.java
package com.yangshengzhou.lucky_sms.pojo;

import com.baomidou.mybatisplus.annotation.*;
import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * 教师实体类
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("teachers")
public class Teacher extends BaseEntity {
    
    @TableId(type = IdType.AUTO)
    private Integer teacherId;
    
    @TableField("teacher_no")
    private String teacherNo;
    
    @TableField("user_id")
    private Integer userId;
    
    @TableField("department_id")
    private Integer departmentId;
    
    @TableField("title_id")
    private Integer titleId;
    
    @JsonFormat(pattern = "yyyy-MM-dd")
    @TableField("hire_date")
    private LocalDate hireDate;
    
    @TableField("office_location")
    private String officeLocation;
    
    private String phone;
    private String email;
    
    @TableField("research_direction")
    private String researchDirection;
    
    @TableField("education_background")
    private String educationBackground;
    
    @TableField("work_experience")
    private String workExperience;
    
    @TableField("status_id")
    private Integer statusId;
    
    // 关联查询字段
    @TableField(exist = false)
    private User user;
    
    @TableField(exist = false)
    private Department department;
    
    @TableField(exist = false)
    private Title title;
    
    @TableField(exist = false)
    private StatusType status;
}

File: backend-go/models/teacher.go
package models

import (
	"time"
)

type Teacher struct {
	BaseEntity
	TeacherID           int          `json:"teacherId" gorm:"column:teacher_id;primaryKey;autoIncrement"`
	TeacherNo           string       `json:"teacherNo" gorm:"column:teacher_no;size:20;uniqueIndex"`
	UserID              int          `json:"userId" gorm:"column:user_id;index"`
	DepartmentID        int          `json:"departmentId" gorm:"column:department_id;index"`
	TitleID             int          `json:"titleId" gorm:"column:title_id;index"`
	HireDate            *time.Time   `json:"hireDate" gorm:"column:hire_date"`
	OfficeLocation      string       `json:"officeLocation" gorm:"column:office_location;size:100"`
	Phone               string       `json:"phone" gorm:"column:phone;size:20"`
	Email               string       `json:"email" gorm:"column:email;size:100"`
	ResearchDirection   string       `json:"researchDirection" gorm:"column:research_direction;type:text"`
	EducationBackground string       `json:"educationBackground" gorm:"column:education_background;type:text"`
	WorkExperience      string       `json:"workExperience" gorm:"column:work_experience;type:text"`
	StatusID            int          `json:"statusId" gorm:"column:status_id"`
	
	User       *User       `json:"user,omitempty" gorm:"-"`
	Department *Department `json:"department,omitempty" gorm:"-"`
	Title      *Title      `json:"title,omitempty" gorm:"-"`
	Status     *StatusType `json:"status,omitempty" gorm:"-"`
}

func (Teacher) TableName() string {
	return "teachers"
}
